egov-location/src/main/java/org/egov/boundary/domain/service/BoundaryService.java

```diff
@@ -330,12 +331,14 @@ public Resource fetchShapeFile(String tenantId) throws IOException {
 		return null;
 	}
 
+	@Cacheable(value = "cBoundariesByIdsAndTypeAndNumberAndCodeAndTenant", sync = true)
 	public List<Boundary> getAllBoundariesByIdsAndTypeAndNumberAndCodeAndTenant(
 			BoundarySearchRequest boundarySearchRequest) {
 
 		return boundaryRepository.getAllBoundariesByIdsAndTypeAndNumberAndCodeAndTenant(boundarySearchRequest);
 	}
-	
+
```

Comment:
Adding spring cache might give problems when we run more than one pod. For example, in case of _update it will evict cache only in the particular pod which handled that particular request. We will have to figure out such use cases and might have to use centralized caching using redis instead to avoid these problems.